Aspose.PDF for .NET 26.4 release notes (#820)
Aspose/releases.aspose.com: content/en/pdf/net/release-notes/2026/aspose-pdf-for-net-26-04-release-notes.md

---
id: "aspose-pdf-for-net-26-4-release-notes"
slug: "aspose-pdf-for-net-26-4-release-notes"
linktitle: "Aspose.PDF for .NET 26.4"
title: "Aspose.PDF for .NET 26.4"
weight: 120
description: "This page contains new Aspose.PDF for .NET features, enhancements, and bug fixes in 2026, version 26.4."
type: "repository"
layout: "release"
family_listing_page_title: "Aspose.PDF for .NET 26.4"
lastmod: "2026-04-14"
sitemap:
  changefreq: "monthly"
  priority: 0.7
---

{{% alert color="primary" %}}

This page contains release notes information for Aspose.PDF for .NET 26.4.

{{% /alert %}}

## Improvements and Changes

### Features and Enhancements

#### Improved PDF to Image Conversion Performance

> Aspose.PDF for .NET 26.4 includes significant performance improvements for PDF to image conversion. This enhancement addresses `PDFNET-41182`, where converting PDF pages to JPEG images had been reported as taking too long, with each page requiring several seconds to process.

Reviewer: ```suggestion
Aspose.PDF for .NET 26.4 introduces significant performance improvements for PDF-to-image conversion. This enhancement addresses issue `PDFNET-41182`, where converting PDF pages to JPEG images was reported to be slow, with each page taking several seconds to process.
```

```csharp
// For complete examples and data files, visit https://github.com/aspose-pdf/Aspose.PDF-for-.NET
private static void ConvertPdfToJpeg()
{
    // The path to the documents directory
    var dataDir = RunExamples.GetDataDir_AsposePdf_DocumentConversion();

    // Open PDF document
    using (var pdfDocument = new Aspose.Pdf.Document(dataDir + "input.pdf"))
    {
        for (int pageCount = 1; pageCount <= pdfDocument.Pages.Count; pageCount++)
        {
            using (var imageStream = new System.IO.FileStream(
                dataDir + string.Format("page{0}.jpg", pageCount),
                System.IO.FileMode.Create))
            {
                // Create Resolution object
                var resolution = new Aspose.Pdf.Devices.Resolution(300);

                // Create JPEG device with specified resolution
                var jpegDevice = new Aspose.Pdf.Devices.JpegDevice(resolution);

                // Convert a particular page and save the image to stream
                jpegDevice.Process(pdfDocument.Pages[pageCount], imageStream);
            }
        }
    }
}
```

#### PDF/UA Structure Tree Processing Performance Improvement

> Aspose.PDF for .NET 26.4 improves the performance of `PdfUaConvertStrategy.Process_7_1_structureTree` for large PDF documents. This enhancement addresses `PDFNET-61884`, where CPU profiling revealed that the `pageMCIDs.Contains(key)` call was taking the most execution time during structure tree processing.

Reviewer: ```suggestion
Aspose.PDF for .NET 26.4 improves the performance of `PdfUaConvertStrategy.Process_7_1_structureTree` when working with large PDF documents. This enhancement addresses issue `PDFNET-61884`, where CPU profiling identified the `pageMCIDs.Contains(key)` call as the primary performance bottleneck during structure tree processing.
```

> By replacing the `List<string>` lookup with a `HashSet<ulong>` and using a composite key instead of string formatting, this update provides a significant performance improvement when converting large PDF documents to PDF/UA format.

Reviewer: ```suggestion
By replacing the `List<string>` lookup with a `HashSet<ulong>` and introducing a composite key instead of relying on string formatting, this update delivers a significant performance boost when converting large PDF documents to PDF/UA format.
```
